tests/drivers/sensor/accel/src/main.c

```diff
@@ -42,24 +42,42 @@ void test_sensor_accel_basic(void)
 }
 
 /* Run all of our tests on an accelerometer device with the given label */
-static void run_tests_on_accel(const struct device *accel)
+static void setup_running_accel(const struct device *accel)
 {
 	zassert_true(device_is_ready(accel), "Accelerometer device is not ready");
 
 	PRINT("Running tests on '%s'\n", accel->name);
 	k_object_access_grant(accel, k_current_get());
+
 	dev = accel;
-	ztest_test_suite(test_sensor_accel,
-			 ztest_user_unit_test(test_sensor_accel_basic));
-	ztest_run_test_suite(test_sensor_accel);
 }
 
-/* test case main entry */
-void test_main(void)
+ZTEST_USER(sensor_accel_0, test_sensor_accel_basic)
+{
+	test_sensor_accel_basic();
+}
+
+static void *accel_0_setup(void)
 {
-	run_tests_on_accel(DEVICE_DT_GET(DT_ALIAS(accel_0)));
+	setup_running_accel(DEVICE_DT_GET(DT_ALIAS(accel_0)));
+
+	return NULL;
+}
+
+ZTEST_SUITE(sensor_accel_0, NULL, accel_0_setup, NULL, NULL, NULL);
 
 #if DT_NODE_EXISTS(DT_ALIAS(accel_1))
-	run_tests_on_accel(DEVICE_DT_GET(DT_ALIAS(accel_1)));
-#endif
+ZTEST_USER(sensor_accel_1, test_sensor_accel_basic)
+{
+	test_sensor_accel_basic();
 }
+
+static void *accel_1_setup(void)
+{
+	setup_running_accel(DEVICE_DT_GET(DT_ALIAS(accel_1)));
+
+	return NULL;
+}
+
+ZTEST_SUITE(sensor_accel_1, NULL, accel_1_setup, NULL, NULL, NULL);
```

Comment:
This is overly complicated, I would much prefer to use a fixture here and a single suite like:

```
struct sensor_accel_fixture {
  const struct device *accel0;
  const struct device *accel1;
};

static void *sensor_accel_setup(void)
{
  static struct sensor_accel_fixture fixture = {
    accel0 = DEVICE_DT_GET(DT_ALIAS(accel_0)),
    accel1 = DEVICE_DT_GET_OR_NULL(DT_ALIAS(accel_1)),
  };
  return &fixture;
}

ZTEST_SUITE(sensor_accel, NULL, sensor_accel_setup, NULL, NULL, NULL);

ZTEST_USER_F(sensor_accel, test_sensor_accel_basic_i2c)
{
  test fixture->accel_0...
}

ZTEST_USER_F(sensor_accel, test_sensor_accel_basic_spi)
{
  if (fixture->accel_1 == NULL) {
    ztest_test_skip();
  }
  ...
}
```

Please note that I forget which node is on i2c and which is on spi (I might have swapped the two)

Reply:
Thanks for the suggestion, it is indeed more simple and we can pass the fixture to test case. Got it fixed, could you please review it again, @yperess , thanks!